### PR TITLE
feat: Allow skipping validation when constructing workspaces

### LINKED ISSUE #1699: Option to skip workspace validation
### Summary

Workspace validation is currently not skippable:

https://github.com/scikit-hep/pyhf/blob/767ed59d06274f810e47944f1fe4c95d94968543/src/pyhf/workspace.py#L289-L298

In some scenarios the validation takes a significant amount of time, and it would be convenient if it was possible to skip it.

### Additional Information

This is using [NormalMeasurement_combined.txt](https://github.com/scikit-hep/pyhf/files/7522251/NormalMeasurement_combined.txt) from https://github.com/scikit-hep/pyhf/discussions/1695. While working with [this gist](https://gist.github.com/alexander-held/2f6727df4cc8f65d75368e293ca83b5a) used in https://github.com/scikit-hep/pyhf/discussions/1695#discussioncomment-1635955, I noticed that workspace validation can take a long time for this example. Below is a minimal reproducer:

```python (paste below)
import pyhf
import json

with open("NormalMeasurement_combined.txt") as f:
    ws = pyhf.Workspace(json.load(f))
```

This runs in around 5.4 seconds for me locally.

When commenting out 

https://github.com/scikit-hep/pyhf/blob/767ed59d06274f810e47944f1fe4c95d94968543/src/pyhf/workspace.py#L298

this time goes down to 1.6 seconds.

The workspace validation takes about as long as the model construction (via `ws.model()`), which for me takes about 4 seconds by itself (again dominated by validation time in this example). It is possible to skip model validation (via `validate=False`), and I believe it would be useful to allow the same for the workspace.

I also noticed that arbitrary kwargs can be passed to workspace construction, e.g. in the above example
```python
ws = pyhf.Workspace(json.load(f), abcdefg=False)
```
also runs fine. It may be convenient to catch these kwargs to help users identify typos.

### Code of Conduct

- [X] I agree to follow the Code of Conduct

### PR DESCRIPTION
**Pull request by alexander-held**
# Description

This adds a `validate` kwarg to the `Workspace` constructor and to `Workspace.build`, replicating the existing functionality in `Model` to skip validation against the JSON schema. For complex workspaces / models, where validation can take an appreciable amount of time (see #1699), this can be a useful option if users are sure that their inputs follow the required format.

There are two things in here that are optional, I tried to split them across commits to easily take out anything you may not want:
- added typing for the `validate` kwargs,
- moved the log info about reference schema to only trigger if validation is actually performed.

There was previously no documentation for `validate` in the `Model` constructor, so this has also been added.

A test for workspace construction without validation is also included, for both construction from a specification and from model + data. This uses the same `simplemodels` setup also used elsewhere in the workspace tests, ~~so this could be refactored if #1709 is merged first (or otherwise could be refactored in #1709). Since #1709 also adds a test at the end, merging one probably results in a conflict with the other PR as well.~~ -> #1709 was merged before, and the implementation here was changed to re-use the fixture added in that PR.

resolves #1699 

# Checklist Before Requesting Reviewer

- [x] Tests are passing
- [x] "WIP" removed from the title of the pull request
- [x] Selected an Assignee for the PR to be responsible for the log summary

# Before Merging

For the PR Assignees:

- [x] Summarize commit messages into a comprehensive review of the PR

```
* Add validate kwarg to pyhf.workspace.Workspace to allow skipping validation
if the user desires.
   - The motivation for this comes from some workspace validations taking
     multiple seconds. Power users might want the ability to turn off
     validation in these cases to save time. This behavior is discouraged
     in general, but should be allowed.
* Add docstring for pyhf.workspace.Workspace and add validate kwarg to
pyhf.pdf.Model docstring.
* Add tests for workspace validate kwarg.
```

### CODE REVIEW

File: src/pyhf/workspace.py [hunk 1]
```diff
@@ -286,16 +286,17 @@ class Workspace(_ChannelSummaryMixin, dict):
 
     valid_joins = ['none', 'outer', 'left outer', 'right outer']
 
-    def __init__(self, spec, **config_kwargs):
+    def __init__(self, spec, validate: bool = True, **config_kwargs):
         """Workspaces hold the model, data and measurements."""
```
Comment (alexander-held) [inline src/pyhf/workspace.py:290]: How does the documentation here work? I am not sure where to document the `validate` kwarg. Should I just expand this existing docstring, and also add `spec`?

File: tests/test_workspace.py [hunk 2]
```diff
@@ -887,3 +887,18 @@ def test_workspace_poiless(datadir):
 
     assert model.config.poi_name is None
     assert model.config.poi_index is None
+
+
+def test_workspace_without_validation(mocker):
+    # TODO: refactor if #1709 is accepted / merged
```
Comment (alexander-held) [inline tests/test_workspace.py:893]: #1709 refactors the `simplemodels` setup to use a fixture instead, so this could be adopted here too for consistency.

**Comment by matthewfeickert** [comment]
Relevant RTD builds:
* [`pyhf.pdf.Model`](https://pyhf--1710.org.readthedocs.build/en/1710/_generated/pyhf.pdf.Model.html#model)
* [`pyhf.workspace.Workspace`](https://pyhf--1710.org.readthedocs.build/en/1710/_generated/pyhf.workspace.Workspace.html#workspace)

File: src/pyhf/pdf.py [hunk 3]
```diff
@@ -635,14 +635,21 @@ class Model:
...
+        **config_kwargs,
     ):
         """
         Construct a HistFactory Model.
 
         Args:
             spec (:obj:`jsonable`): The HistFactory JSON specification
-            batch_size (:obj:`None` or :obj:`int`): Number of simultaneous (batched) Models to compute.
+            batch_size (:obj:`None` or :obj:`int`): Number of simultaneous (batched)
+            Models to compute.
```
Comment (matthewfeickert) [inline src/pyhf/pdf.py:651]:
```suggestion
             Models to compute.
```

space is just for alignment issues in docs

File: src/pyhf/workspace.py (hunk 1 above)
Comment (matthewfeickert) [reply src/pyhf/workspace.py:290]: Yeah, the docstrings get a bit fincky for the classes. What you suggest sounds great, and you can take a look at https://scikit-hep.org/pyhf/_generated/pyhf.patchset.PatchSet.html for an example of how we've got things kinda working now.

File: tests/test_workspace.py (hunk 2 above)
Comment (matthewfeickert) [reply tests/test_workspace.py:893]: Yup, this all sounds good. We can merge PR #1709 first and then rebase this one on top.

**Review by matthewfeickert — COMMENTED** [review]
Thanks for the PR @alexander-held. This all is looking good to me and your plan for docstrings sounds great. :+1:

File: src/pyhf/workspace.py (hunk 1 above)
Comment (alexander-held) [reply src/pyhf/workspace.py]: I've updated the docstring, it's also quite similar to the one for the `Model` constructor.

**Comment by matthewfeickert** [comment]
Okay this one can get rebased and refactored now. :+1:

File: tests/test_workspace.py (hunk 2 above)
Comment (alexander-held) [reply tests/test_workspace.py]: Done now in e737e4d.

**Review by matthewfeickert — APPROVED** [review]

**Review by kratsg — APPROVED** [review]

**Comment by kratsg** [comment]
Congrats on the 1k'th commit @alexander-held :)

**Pull request merged**